auth/postgres/repo.go

```diff
@@ -0,0 +1,558 @@
+		return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+	if cnt == 0 {
+		return auth.PAT{}, repoerr.ErrNotFound
+	}
+
+	pat, err := pr.retrieveFromDB(ctx, userID, patID)
+	if err != nil {
+		return auth.PAT{}, err
+	}
+
+	if err := pr.cache.Save(ctx, patID, pat); err != nil {
+		return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	return pat, nil
+}
+
+func (pr *patRepo) Revoke(ctx context.Context, userID, patID string) error {
+	q := `
+		UPDATE pats 
+		SET revoked = true, revoked_at = $1
+		WHERE user_id = $2 AND id = $3`
+
+	res, err := pr.db.ExecContext(ctx, q, time.Now(), userID, patID)
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+	if cnt == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	if err := pr.cache.Remove(ctx, patID); err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	return nil
+}
+
+func (pr *patRepo) Reactivate(ctx context.Context, userID, patID string) error {
+	q := `
+		UPDATE pats 
+		SET revoked = false, revoked_at = NULL
+		WHERE user_id = $1 AND id = $2`
+
+	res, err := pr.db.ExecContext(ctx, q, userID, patID)
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+	if cnt == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	pat, err := pr.Retrieve(ctx, userID, patID)
+	if err != nil {
+		return err
+	}
+
+	if err := pr.cache.Save(ctx, patID, pat); err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	return nil
+}
+
+func (pr *patRepo) Remove(ctx context.Context, userID, patID string) error {
+	q := `DELETE FROM pats WHERE user_id = $1 AND id = $2`
+
+	res, err := pr.db.ExecContext(ctx, q, userID, patID)
+	if err != nil {
+		return postgres.HandleError(repoerr.ErrRemoveEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(repoerr.ErrRemoveEntity, err)
+	}
+	if cnt == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	if err := pr.cache.Remove(ctx, patID); err != nil {
+		return errors.Wrap(repoerr.ErrRemoveEntity, err)
+	}
+
+	return nil
+}
+
+func (pr *patRepo) AddScopeEntry(ctx context.Context, userID, patID string, platformEntityType auth.PlatformEntityType, optionalDomainID string, optionalDomainEntityType auth.DomainEntityType, operation auth.OperationType, entityIDs ...string) (auth.Scope, error) {
+	pat, err := pr.retrieveFromDB(ctx, userID, patID)
+	if err != nil {
+		return auth.Scope{}, err
+	}
+
+	if err := pat.Scope.Add(platformEntityType, optionalDomainID, optionalDomainEntityType, operation, entityIDs...); err != nil {
+		return auth.Scope{}, errors.Wrap(repoerr.ErrCreateEntity, err)
+	}
+
+	_, scope, err := patToDBRecords(pat)
```

Comment:
Please use `toDBScope` which will return only scope
Signature of the function will be `func toDBScope(auth.PAT) (dbScope,error)`